pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -34,3 +62,133 @@ message BoundContract {
   string name = 2;
   bool pending = 3;
 }
+
+enum ComparisonOperator {
+  Eq = 0;
+  Neq = 1;
+  Gt = 2;
+  Lt = 3;
+  Gte = 4;
+  Lte = 5;
+}
+
+message QueryKeyFilter {
+  string key = 1;
+  // Expressions are lightweight orm like DSL defined for filtering over common blockchain primitives.
+  repeated Expression expression = 2;
+}
+
+// Expression encapsulates a single unit of filtering logic, which can be a common blockchain primitive or a composite of boolean expressions. 
+// This allows for both simple and more complex nested expressions.
+message Expression {
+  oneof evaluator {
+    Primitive primitive = 1;
+    BooleanExpression boolean_expression = 2;
+  }
+}
+
+enum BooleanOperator {
+  AND = 0;
+  OR = 1;
+}
+
+message BooleanExpression {
+  BooleanOperator boolean_operator = 1;
+  repeated Expression expression = 2;
+}
+
+message And {
+  repeated Expression expr = 1;
+}
+
+message Or {
+  repeated Expression expr = 1;
+}
+
+message ValueComparator  {
+  string  value = 1;
+  ComparisonOperator operator = 2;
+}
+
+message Comparator  {
+  string name = 1;
+  repeated ValueComparator value_comparators = 2;
+}
+
+message Block{
+  uint64 block_number = 1;
+  ComparisonOperator operator = 2;
+}
+
+enum ConfirmationLevel {
+  Finalized   = 0;
```

Comment:
We changed the defaults for these. Should they or can they be changed here too to match? Not sure what can be done in protobufs with negative numbers.